### PR TITLE
[KOGITO-4619] - Prepare the kogito-images repository for productization

### PR DESCRIPTION
**Pull request by spolti**
Signed-off-by: spolti <fspolti@redhat.com>
See: https://issues.redhat.com/browse/KOGITO-4619

Many thanks for submitting your Pull Request :heart:! 

Please make sure your PR meets the following requirements:

- [x] You have read the [contributors guide](README.md#contributing-to-kogito-images-repository)
- [x] Pull Request title is properly formatted: `[KOGITO-XYZ] Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket
- [x] Your feature/bug fix has a testcase that verifies it
- [ ] You've tested the new feature/bug fix in an actual OpenShift cluster

### CODE REVIEW

File: scripts/list-images.py [hunk 1]
```diff
@@ -1,15 +1,34 @@
...
-sys.dont_write_bytecode = True
 
 import common
 
+sys.dont_write_bytecode = True
+
+PRODUCT_PREFIX = "rhpam"
+
+# map community images with its correlative product
+images_map = {'rhpam-kogito-builder-rhel8': 'kogito-builder',
```
Comment (radtriste) [inline scripts/list-images.py]: I guess if we have a convention, we could make that mapping more dynamic.
`prod image = rhpam-{community-image-name}-rhel8`

File: scripts/list-images.py [hunk 2]
```diff
@@ -1,15 +1,34 @@
...
-    for img in images:
-        print(img)
+    if args.parent_image:
+        print(images_map[args.parent_image])
+    else:
+        for img in sorted(common.get_all_images()):
+            if img.startswith(PRODUCT_PREFIX) and args.prod:
+                print(img)
+            elif not img.startswith(PRODUCT_PREFIX) and not args.prod:
+                print(img)
```
Comment (radtriste) [inline scripts/list-images.py]: default case is do not print anything right ?

I would find this more readable:
```suggestion
            if args.prod:
                if img.startswith(PRODUCT_PREFIX):
                    print(img)
            else:
                if not img.startswith(PRODUCT_PREFIX):
                    print(img)
```

**Review by radtriste — COMMENTED** [review]
I don't see any feature tests for rhpam builder and runtime ... is that on purpose ?

**Comment by spolti** [comment]
> I don't see any feature tests for rhpam builder and runtime ... is that on purpose ?

Yeah, I didn't added too much yet, there is a email thread where we are discussing how we will be approaching the productization, this is the first draft.

File: scripts/list-images.py (hunk 1 above)
Comment (spolti) [reply scripts/list-images.py]: yeah, good idea, will try that.

File: scripts/list-images.py (hunk 2 above)
Comment (spolti) [reply scripts/list-images.py]: +1

**Comment by radtriste** [comment]
There is an ongoing problem with examples due to conflict between merge and nightly run ... 
Nightly has been restarted to redeploy artifacts correctly

**Comment by radtriste** [comment]
Jenkins retest this

**Comment by radtriste** [comment]
@spolti you need to add `kogito-common-builder.feature` to that line => https://github.com/kiegroup/kogito-images/blob/master/scripts/common.py#L34
else we don't ignore certificate in those tests and it fails

**Comment by spolti** [comment]
@radtriste nice catch, I completely forgot about it. thanks.

File: tests/features/kogito-common-builder.feature [hunk 3]
```diff
@@ -0,0 +1,436 @@
+@quay.io/kiegroup/kogito-builder @rhpam-7/rhpam-kogito-builder-rhel8
+Feature: kogito-builder image tests
+
+  Scenario: Verify if the s2i build is finished as expected using native build and runtime image
```
Comment (radtriste) [inline tests/features/kogito-common-builder.feature]: I thought we would not have yet the `native` supported in rhpam-7/rhpam-kogito-builder-rhel8
do we still want to test it ?

Comment (radtriste) [reply tests/features/kogito-common-builder.feature]: or should we create a `kogito-builder-native.feature` file for the community version only ?

**Review by radtriste — COMMENTED** [review]
looks ok. See my last comments for the testing of product image

File: tests/features/kogito-common-builder.feature (hunk 3 above)
Comment (spolti) [reply tests/features/kogito-common-builder.feature]: Nice catch man :)
yeah, let's create a new for native tests, best keep them apart as well :)

Comment (spolti) [reply tests/features/kogito-common-builder.feature]: updated, please take a look.

File: tests/features/kogito-common-builder-jvm.feature [hunk 4]
```diff
@@ -0,0 +1,304 @@
...
+      | wait                 | 80                                                                            |
+      | request_method       | POST                                                                          |
+      | request_body         | {"approver" : "john", "order" : {"orderNumber" : "12345", "shipped" : false}} |
+      | content_type         | application/json                                                              |
+      | expected_status_code | 201                                                                           |
+    And file /home/kogito/bin/process-springboot-example.jar should exist
+    And container log should contain main] .c.l.ClasspathLoggingApplicationListener
+    And run sh -c 'echo $JAVA_OPTIONS' in container and immediately check its output for -Ddebug=true
+
+  Scenario: Perform a incremental s2i build
```
Comment (radtriste) [inline tests/features/kogito-common-builder-jvm.feature]: I see 3 scenarios with name `Perform a incremental s2i build` in builder feature files. 
Can you make sure scenario names are unique ? maybe add `native` and `quarkus/springboot` information in the name ?

**Review by radtriste — COMMENTED** [review]
nice work. One last comment

File: tests/features/kogito-common-builder-jvm.feature (hunk 4 above)
Comment (spolti) [reply tests/features/kogito-common-builder-jvm.feature]: Oops, forget to update the scenario name, thanks for pointing out.

Comment (spolti) [reply tests/features/kogito-common-builder-jvm.feature]: updated, let me know wdyt about the new scenario's names.

File: tests/features/kogito-builder-native.feature [hunk 5]
```diff
@@ -0,0 +1,140 @@
+@quay.io/kiegroup/kogito-builder
+Feature: kogito-builder image native build tests
+
+  Scenario: verify java cacerts and libsunec are available in the given container.
+    When container is started with command bash
+    Then  file /home/kogito/ssl-libs/libsunec.so should exist
+    And file /home/kogito/cacerts should exist
+
+  Scenario: verify if the maven and graal vm settings are correct
```
Comment (radtriste) [inline tests/features/kogito-builder-native.feature]: `verify if the maven and graal vm settings are correct` is appearing also in `kogito-common-builder-jvm.feature`

Can you make sure there is no duplicate scenario names between the different feature files ?

**Review by radtriste — CHANGES_REQUESTED** [review]

**Review by radtriste — APPROVED** [review]
nice work

**Review by msmagnanijr — APPROVED** [review]
@spolti  nice work! lgtm.

File: content_sets.yaml [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+#
+# See https://mojo.redhat.com/docs/DOC-1023066 for more information on
+# maintaining this file and the format and examples
+#
+# You should have one top level item for each architecture being built. Most
+# likely this will be x86_64 and ppc64le initially.
+---
+x86_64:
+  - rhel-8-for-x86_64-baseos-rpms
+  - rhel-8-for-x86_64-appstream-rpms
```
Comment (ricardozanini) [inline content_sets.yaml:16]: Any of these have `jq`? https://github.com/kiegroup/kogito-images/blob/master/modules/kogito-jq/module.yaml#L15

File: rhpam-kogito-builder-rhel8-overrides.yaml [hunk 7]
```diff
@@ -0,0 +1,25 @@
+schema_version: 1
+
+name: "rhpam-7/rhpam-kogito-builder-rhel8"
+version: "7.11.0"
+from: "registry.redhat.io/ubi8/ubi-minimal:latest"
+description: "RHPAM Platform for building Kogito based on Quarkus or Springboot"
```
Comment (ricardozanini) [inline rhpam-kogito-builder-rhel8-overrides.yaml]:
```suggestion
description: "RHPAM Platform for building Kogito based on Quarkus or SpringBoot"
```

File: rhpam-kogito-builder-rhel8-overrides.yaml [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+
+labels:
+  - name: "com.redhat.component"
+    value: "rhpam-7-kogito-runtime-jvm-rhel8-container"
+  - name: "io.k8s.display-name"
+    value: "Red Hat build of Kogito based on Quarkus or Spring Boot"
```
Comment (ricardozanini) [inline rhpam-kogito-builder-rhel8-overrides.yaml]:
```suggestion
    value: "Red Hat build of Kogito based on Quarkus or SpringBoot"
```

File: rhpam-kogito-imagestream.yaml [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+  annotations:
+    description: ImageStream definitions for Red Hat build of Kogito images
+    openshift.io/provider-display-name: Kie Group.
+items:
+  - kind: ImageStream
+    apiVersion: v1
+    metadata:
+      name: rhpam-kogito-runtime-jvm-rhel8
+      annotations:
+        openshift.io/display-name: Runtime image for Kogito based on Quarkus or Springboot JVM image
```
Comment (ricardozanini) [inline rhpam-kogito-imagestream.yaml]:
```suggestion
        openshift.io/display-name: Runtime image for Kogito based on Quarkus or SpringBoot JVM image
```

File: rhpam-kogito-imagestream.yaml [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+        openshift.io/provider-display-name: Kie Group.
+    spec:
+      tags:
+        - name: '7.11.0'
+          annotations:
+            description: Runtime image for Kogito based on Quarkus or Springboot JVM image
```
Comment (ricardozanini) [inline rhpam-kogito-imagestream.yaml]:
```suggestion
            description: Runtime image for Kogito based on Quarkus or SpringBoot JVM image
```

File: rhpam-kogito-imagestream.yaml [hunk 11]
```diff
@@ -0,0 +1,50 @@
...
+            type: Local
+          from:
+            kind: DockerImage
+            name: registry.redhat.io/rhpam-7/rhpam-kogito-runtime-jvm-rhel8:7.11.0
+  - kind: ImageStream
+    apiVersion: v1
+    metadata:
+      name: rhpam-kogito-builder-rhel8
+      annotations:
+        openshift.io/display-name: Platform for building Kogito based on Quarkus or Springboot
```
Comment (ricardozanini) [inline rhpam-kogito-imagestream.yaml]:
```suggestion
        openshift.io/display-name: Platform for building Kogito based on Quarkus or SpringBoot
```

File: rhpam-kogito-imagestream.yaml [hunk 12]
```diff
@@ -0,0 +1,50 @@
...
+        openshift.io/provider-display-name: Kie Group.
+    spec:
+      tags:
+        - name: '7.11.0'
+          annotations:
+            description: Platform for building Kogito based on Quarkus or Springboot
```
Comment (ricardozanini) [inline rhpam-kogito-imagestream.yaml]:
```suggestion
            description: Platform for building Kogito based on Quarkus or SpringBoot
```

File: rhpam-kogito-runtime-jvm-rhel8-overrides.yaml [hunk 13]
```diff
@@ -0,0 +1,25 @@
+schema_version: 1
+
+name: "rhpam-7/rhpam-kogito-runtime-jvm-rhel8"
+version: "7.11.0"
+from: "registry.redhat.io/ubi8/ubi-minimal:latest"
+description: "RHPAM Runtime image for Kogito based on Quarkus or Springboot JVM image"
```
Comment (ricardozanini) [inline rhpam-kogito-runtime-jvm-rhel8-overrides.yaml]:
```suggestion
description: "RHPAM Runtime image for Kogito based on Quarkus or SpringBoot JVM image"
```

File: rhpam-kogito-runtime-jvm-rhel8-overrides.yaml [hunk 14]
```diff
@@ -0,0 +1,25 @@
...
+
+labels:
+  - name: "com.redhat.component"
+    value: "rhpam-7-kogito-runtime-jvm-rhel8-container"
+  - name: "io.k8s.display-name"
+    value: "Red Hat build of Kogito based on Quarkus or Spring Boot JVM image"
```
Comment (ricardozanini) [inline rhpam-kogito-runtime-jvm-rhel8-overrides.yaml]:
```suggestion
    value: "Red Hat build of Kogito based on Quarkus or SpringBoot JVM image"
```

**Review by ricardozanini — APPROVED** [review]
Just a heads-up regarding `jq`. Since I think we removed the feature that required it, we should be fine removing this module. :thinking:

**Comment by spolti** [comment]
> Just a heads-up regarding `jq`. Since I think we removed the feature that required it, we should be fine removing this module.

I think so, we could address it on another jira.

**Comment by spolti** [comment]
have opened https://issues.redhat.com/browse/KOGITO-4717

File: content_sets.yaml (hunk 6 above)
Comment (spolti) [reply content_sets.yaml:16]: I think so, at least the build using Red Hat infra, didn't fail :)

**Comment by spolti** [comment]
@ricardozanini thanks for the SpringBoot renaming suggestions :D

**Review by antncl — APPROVED** [review]
Forgot to add my approval, I did a limited review of the bits i'm familiar with (osbs, content sets and gating)

**Review by almope — APPROVED** [review]

**Comment by spolti** [comment]
thanks.

**Pull request merged**